packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+
+#import "FVPAVFactory.h"
+#import "FVPDisplayLink.h"
+#import "FVPFrameUpdater.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// FVPVideoPlayer is responsible for managing video playback using AVPlayer.
+/// It provides methods to control playback, adjust volume, handle seeking, and
+/// notify the Flutter engine about new video frames.
+@interface FVPVideoPlayer : NSObject
+/// The Flutter event channel used to communicate with the Flutter engine.
+@property(nonatomic) FlutterEventChannel *eventChannel;
```

Comment:
Pulling this into a separate file really highlights some of the problematic coupling between the plugin class and the player class :(

(No action needed here, just commenting for future `git blame` archeologists wondering why this is a public read-write property.)